BDArmory/Control/MissileFire.cs

```diff
@@ -5228,6 +5226,8 @@ bool SmartPickWeapon_EngagementEnvelope(TargetInfo target)
                             {
                                 if (targetWeaponRPM < candidateRPM)
                                 {
+                                    if ((targetWeapon != null) && (targetWeapon.GetWeaponClass() == WeaponClasses.Missile) && (targetWeaponTDPS > 0))
```

Comment:
Presumably, this should also apply for rockets and lasers (which doesn't even have this check)?

Also, it's a minor optimisation, but this series of `if (candidateClass == ...)` checks could be replaced with a `switch (item.Current.GetWeaponClass())` statement.